File: pypp_cli/test_dir/python/src/number_types/number_types.py
from pypp_python import (
    int8_t,
    int32_t,
    int64_t,
    int16_t,
    uint8_t,
    uint16_t,
    uint32_t,
    uint64_t,
    float32,
)


def number_types_fn():
    print("NUMBER TYPES RESULTS:")
    a: int8_t = -1
    b: int16_t = -129
    c: int32_t = -32769
    d: int64_t = -2147483649
    e: uint8_t = 1
    f: uint16_t = 256
    g: uint32_t = 65536
    h: uint64_t = 4294967296
    print(a, b, c, d, e, f, g, h)
    i: float32 = 3.14
    print(i)

File: pypp_cli/test_dir/cpp/src/number_types/number_types.cpp
#include "src/number_types/number_types.h"
#include "py_str.h"
#include "pypp_util/print.h"
#include <cstdint>

namespace me {
void number_types_fn() {
    pypp::print(pypp::PyStr("NUMBER TYPES RESULTS:"));
    int8_t a = -1;
    int16_t b = -129;
    int32_t c = -32769;
    int64_t d = -2147483649;
    uint8_t e = 1;
    uint16_t f = 256;
    uint32_t g = 65536;
    uint64_t h = 4294967296;
    pypp::print(a, b, c, d, e, f, g, h);
    float i = 3.14;
    pypp::print(i);
}

} // namespace me